### PR TITLE
kconf: refactor

### PR DESCRIPTION
**Pull request by Sail0rd**
## Description of changes

Changed directory to respect new packages convention + added completions for common shells

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/kc/kconf/package.nix [hunk 1]
```diff
@@ -14,14 +18,25 @@ buildGoModule rec {
...
+      --zsh <($out/bin/kconf completion zsh)
+  '';
+
   meta = with lib; {
     description = "Opinionated command line tool for managing multiple kubeconfigs";
     mainProgram = "kconf";
     homepage = "https://github.com/particledecay/kconf";
     license = licenses.mit;
-    maintainers = with maintainers; [ thmzlt ];
+    maintainers = with maintainers; [thmzlt sailord vinetos];
```
Comment (Vinetos) [inline pkgs/by-name/kc/kconf/package.nix:40]:
```suggestion
    maintainers = with maintainers; [ thmzlt sailord vinetos];
```

**Review by Vinetos — CHANGES_REQUESTED** [review]
looks good otherwise

File: pkgs/by-name/kc/kconf/package.nix (hunk 1 above)
Comment (Sail0rd) [reply pkgs/by-name/kc/kconf/package.nix:40]: The space was removed by my formatter 😅 (alejandra) do you really think I should put a space here ?

Comment (Vinetos) [reply pkgs/by-name/kc/kconf/package.nix]: Can you try with `nix run nixpkgs#nixfmt-rfc-style package.nix` ?

Comment (Sail0rd) [reply pkgs/by-name/kc/kconf/package.nix]: just did, it should be fine now 👍

File: pkgs/by-name/kc/kconf/package.nix [hunk 2]
```diff
@@ -14,14 +17,25 @@ buildGoModule rec {
...
   ldflags = [
-      "-s" "-w" "-X github.com/particledecay/kconf/build.Version=${version}"
+    "-s"
+    "-w"
+    "-X github.com/particledecay/kconf/build.Version=${version}"
   ];
 
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
```
Comment (pbsds) [inline pkgs/by-name/kc/kconf/package.nix:27]:
```suggestion
  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
```

Comment (Sail0rd) [reply pkgs/by-name/kc/kconf/package.nix:27]: what are you suggesting exactly ? I don't really see the purpose of this 😅

**Comment by pbsds** [comment]
formatting error

**Comment by Sail0rd** [comment]
I don't understand i tried every formatter i know of, what's causing the test to fail ?

**Comment by Sail0rd** [comment]
turns out I had to format it using `nixfmt-rfc-style` :/

**Review by Vinetos — APPROVED** [review]

**Pull request merged**